bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusBistabielHandler.java

```diff
@@ -0,0 +1,214 @@
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     * Returns the serial number of the controller
+     *
+     * @return the serial nr
+     */
+    public @Nullable String getSN() {
+        return this.sn;
+    }
+
+    /**
+     * Sets the serial number of the controller
+     */
+    public void setSN() {
+        QbusBridgeHandler QBridgeHandler = getBridgeHandler("Bistabiel", bistabielId);
+        if (QBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No communication with Qbus Bridge!");
+            return;
+        } else {
+            this.sn = QBridgeHandler.getSn();
+        }
+    }
+
+    /**
+     * Handle the status update from the thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication QComm = getCommunication("Bistabiel", bistabielId);
+        if (QComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for bistabiel " + bistabielId);
+            return;
+        }
+
+        Map<Integer, QbusBistabiel> bistabielComm = QComm.getBistabiel();
+
+        if (bistabielComm != null) {
+            QbusBistabiel QBistabiel = bistabielComm.get(bistabielId);
+
+            if (QBistabiel == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for bistabiel "
+                                + bistabielId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!QComm.communicationActive()) {
+                        restartCommunication(QComm, "Bistabiel", bistabielId);
+                    }
+
+                    if (QComm.communicationActive()) {
+
+                        if (command == REFRESH) {
+                            handleStateUpdate(QBistabiel);
+                            return;
+                        }
+
+                        handleSwitchCommand(QBistabiel, channelUID, command);
+                    }
+
+                });
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     * Executes the switch command
+     */
+    private void handleSwitchCommand(QbusBistabiel QBistabiel, ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            OnOffType s = (OnOffType) command;
+            @Nullable
+            String snr = getSN();
```

Comment:
The compiler is more intelligent on local variables. Therefore, the Nullable could be removed. Please check all.
```suggestion
            String snr = getSN();
```